app/workers/my_anime_list_sync_worker.rb

```diff
@@ -1,10 +1,20 @@
 class MyAnimeListSyncWorker
   include Sidekiq::Worker
 
-  def perform(library_entry_id, method)
+  # data is a hash
+  def perform(data)
```

Comment:
ok i'll rephrase what I said, It is identical where needed. 
`media = MyAnimeListSyncService.new(library_entry, data['method'])`
`
`library_entry` is going to be for delete
```
 user_id: 1,
 media_id: 3936,
 media_type: "Anime",
 method: "delete"
 create/update:
```
 id: 1,
 user_id: 1,
 media_id: 3936,
 status: 1,
 created_at: Thu, 19 Jan 2017 19:24:07 UTC +00:00,
 updated_at: Thu, 19 Jan 2017 19:32:30 UTC +00:00,
 progress: 4,
 rating: nil,
 private: false,
 notes: nil,
 reconsume_count: 0,
 reconsuming: false,
 media_type: "Anime",
 volumes_owned: 0,
 nsfw: false,
 anime_id: 3936,
 manga_id: nil,
 drama_id: nil>
```

so the only difference is that the method is inside the hash but it isn't even used (I can change it so we do `perform(data, method)` if that will make you happier?